autoconfigure/collector-sqs/src/main/java/zipkin/autoconfigure/collector/sqs/ZipkinSQSCollectorAutoConfiguration.java

```diff
@@ -37,8 +42,28 @@
   /** By default, get credentials from the {@link DefaultAWSCredentialsProviderChain */
   @Bean
   @ConditionalOnMissingBean
-  AWSCredentialsProvider credentialsProvider() {
-    return new DefaultAWSCredentialsProviderChain();
+  AWSCredentialsProvider credentialsProvider(ZipkinSQSCollectorProperties properties) {
+    // Create credentials provider from ID and secret if given otherwise use the default provider
+    AWSCredentialsProvider provider;
+    if (!isNullOrEmpty(properties.awsAccessKeyId) && !isNullOrEmpty(properties.awsSecretAccessKey)) {
+      provider = new AWSStaticCredentialsProvider(
+          new BasicAWSCredentials(properties.awsAccessKeyId, properties.awsSecretAccessKey));
+    } else {
+      provider = new DefaultAWSCredentialsProviderChain();
+    }
+
+    // Create STS provider if a role is specified
+    if (!isNullOrEmpty(properties.awsStsRoleArn)) {
```

Comment:
ok so in this case, instead of using a pre-specified token, we are looking one up at runtime (potentially refreshing it). It might be worthwhile to inject this service, for example as an autowired, required=false field. Then we provide the service with `@ConditionalOnProperty`. in this code block, we'd check if that autowired field was null or not. The advantage to this is that one could handle shutdown more gracefully, and also any problems starting it up wouldn't be nested in this provider. Make sense?